src/types/index.ts

```diff
@@ -0,0 +1,173 @@
+	 */
+	message?: string
+	/**
+	 * True if the message hide automatically, false if the user should have to interact with it
+	 */
+	autoHide?: boolean
+	/**
+	 * Milliseconds before autoHide happens
+	 */
+	visibilityTime?: number
+	/**
+	 * Height of the toast component
+	 */
+	height?: number
+	/**
+	 * Points from the top of the screen (when position='top')
+	 */
+	topOffset?: number
+	/**
+	 * Points from the bottom of the screen (when position='bottom')
+	 */
+	bottomOffset?: number
+	/**
+	 * friction value for the show/hide animation
+	 */
+	friction?: number
+	/**
+	 * Callback when a toast is shown
+	 */
+	onShow?: (toast: ToastProps) => void
+	/**
+	 * Callback when a toast is queued
+	 */
+	onQueue?: (toast: ToastProps) => void
+	/**
+	 * Callback when a toast is hidden (automatically or by the user)
+	 */
+	onHide?: (toast: ToastProps) => void
+	/**
+	 * Callback when a toast is pressed (prevents onHide from firing)
+	 */
+	onPress?: (toast: ToastProps) => void
+}
+
+export interface ToastProviderProps {
+	/**
+	 * Optionally provide defaults for all toasts (settings here can be overridden when showing or queueing a toast)
+	 */
+	defaults?: ToastProps
+	/**
+	 * Optionally override default colors
+	 */
+	colors?: ToastColors
+	/**
+	 * Configure your own toast components to render. Object keys maps to the `ToastProps` `type` prop while the values are render functions to render the toast.
+	 */
+	customToasts?: ToastComponentsConfig
+	/**
+	 * The Toaster will be rendered for you by default unless you set this to false.
+	 * If you need to render it in a place other than where the provider is placed the useToast hook will give you the Toaster you need to render.
+	 * react-navigation is known to interfere with the rendering of the toasts. Place the Toaster just before your </NavigationContainer> to fix it.
+	 */
+	renderToaster?: boolean
+}
+
+export interface UseToastHook {
+	/**
+	 * Pops up a toast (jumps to the front of the queue and hides the existing toast if there is one)
+	 */
+	showToast: (props: ToastProps) => void
+	/**
+	 * Queues up a toast (or shows it instantly if there are no toasts showing and in the queue)
+	 */
+	queueToast: (props: ToastProps) => void
+	/**
+	 * Hides the current toast, which will show the next toast in the queue if there is one
+	 */
+	hideToast: () => void
+	/**
+	 * Purges the toast queue (but does not hide the one being shown)
+	 */
+	clearToastQueue: () => void
+}
+
+export interface IToastContext extends ToastContextSettings, UseToastHook {
+	toasts: Readonly<ToastProps[]>
+	activeToast?: ToastProps | null
+}
+
+type OptionalSettings = Pick<
+	ToastProps,
+	'title' | 'message' | 'onShow' | 'onQueue' | 'onHide' | 'onPress'
+>
+
+interface ToastContextSettings {
+	colors: Required<ToastColors>
+	customToasts?: ToastComponentsConfig
+	defaults: Required<Omit<ToastProps, keyof OptionalSettings>> & OptionalSettings
+}
+
+export type Color = ColorValue & SvgColor
+export type ToastColors = {
```

Comment:
From what I've read `interface` is preferable over `type` when possible. There's a lint rule out there for this. https://ncjamieson.com/prefer-interfaces/

Reply:
Nice. I looked into when to use each as they're confusingly similar. I started with types for everything and moved to interfaces as I needed to extend them, which I found easier to do with interfaces than unioning types. The guidance I found was basically "do what you want", which wasn't very helpful. It's good to know there's a distinct difference in how they behave.